app/widgets/common/secrets/SecretActions.ts

```diff
@@ -17,6 +19,15 @@ export type Secret = {
     provider_name: string;
     provider_options: string;
 };
+
+interface SecretRequestBody {
+    key: string;
+    value: string;
+    visibility: Visibility;
+    is_hidden_value: boolean;
+    provider_name?: string;
+    provider_options?: ProviderOptions;
+}
```

Comment:
Optional:

If you introduced it here, then I think we should be more precise. I went through https://docs.cloudify.co/api/staging/dev/#create-secret and https://docs.cloudify.co/api/staging/dev/#update-secret, so WDYT about making it more accurate (in terms of fields presence and required/optional indicator):
```suggestion
type CreateSecretRequestBody = Pick<Secret, 'value'> & Optional<Pick<Secret, 'visibility' | 'is_hidden_value' | 'schema' | 'provider_name' | 'provider_options'>>
```
There's also `update_if_exists`, but I think we don't have to include it for now.

If we are introducing it for create, then probably we could also do it for update:
```
type UpdateSecretRequestBody = Omit<CreateSecretRequestBody, 'schema'>;
```
There's `creator` also, but similarly to `update_if_exists` I don't think we need it now.

Reply:
Updated my previous comment. Not `Optional`, but `Partial` 🙂 Sorry.